components/dashboards/org.wso2.carbon.dashboards.siddhi.apps.api/src/main/java/org/wso2/carbon/siddhi/apps/api/rest/impl/SiddhiAppsApiServiceImpl.java

```diff
@@ -0,0 +1,205 @@
+import com.google.gson.reflect.TypeToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.analytics.permissions.PermissionProvider;
+import org.wso2.carbon.analytics.permissions.bean.Permission;
+import org.wso2.carbon.siddhi.apps.api.rest.SiddhiAppsApiService;
+import org.wso2.carbon.siddhi.apps.api.rest.bean.SiddhiAppContent;
+import org.wso2.carbon.siddhi.apps.api.rest.bean.SiddhiStoreElement;
+import org.wso2.carbon.siddhi.apps.api.rest.internal.SiddhiAppsDataHolder;
+import org.wso2.carbon.siddhi.apps.api.rest.worker.WorkerServiceFactory;
+import org.wso2.msf4j.Request;
+import org.wso2.siddhi.query.api.SiddhiApp;
+import org.wso2.siddhi.query.compiler.SiddhiCompiler;
+import java.io.IOException;
+import java.io.Reader;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.siddhi.apps.api.rest.utils.Constants.AGGREGATION;
+import static org.wso2.carbon.siddhi.apps.api.rest.utils.Constants.PROTOCOL;
+import static org.wso2.carbon.siddhi.apps.api.rest.utils.Constants.STORE_ANNOTATION;
+import static org.wso2.carbon.siddhi.apps.api.rest.utils.Constants.TABLE;
+import static org.wso2.carbon.siddhi.apps.api.rest.utils.Constants.WINDOW;
+
+/**
+ * Implementation of SiddhiApps REST API
+ */
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaMSF4JServerCodegen",
+        date = "2017-11-01T11:26:25.925Z")
+public class SiddhiAppsApiServiceImpl extends SiddhiAppsApiService {
+
+    private static final Logger log = LoggerFactory.getLogger(SiddhiAppsApiServiceImpl.class);
+    private static final String PERMISSION_APP_NAME = "DASH";
+    private static final String VIEW_SIDDHI_APP_PERMISSION_STRING = "DASH.siddhiApp.viewer";
+    private static final Type listType = new TypeToken<List<String>>() { }.getType();
+    private Gson gson = new Gson();
+    private PermissionProvider permissionProvider;
+
+    public SiddhiAppsApiServiceImpl() {
+        permissionProvider = SiddhiAppsDataHolder.getInstance().getPermissionProvider();
+    }
+
+    @Override
+    public Response getSiddhiAppStoreElements(Request request, String appName) {
+        if (getUserName(request) != null && !permissionProvider.hasPermission(getUserName(request), new Permission
+                (PERMISSION_APP_NAME, VIEW_SIDDHI_APP_PERMISSION_STRING))) {
+            return Response.status(Response.Status.UNAUTHORIZED).entity("Insufficient permissions to view Siddhi " +
+                    "Apps for user " + getUserName(request)).build();
+        }
+
+        if (SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().isEmpty()) {
+            getSiddhiAppsFromWorkers();
+        }
+
+        for (Map.Entry<String, List<SiddhiStoreElement>> entry :
+                SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().entrySet()) {
```

Comment:
```suggestion
                SiddhiAppsDataHolder.getInstance().getSiddhiAppMap().get(appName)) {
```
Use a simple get, we do not need to iterate through all siddhi app entries

Reply:
Fixed in 87c49f3